tests/test_dmxrange_add_sub.py

```diff
@@ -0,0 +1,47 @@
+"""Test functions for adding and removing DMX ranges.""" 
+
+import pytest
+
+from pint.models.dispersion_model import DispersionDM, DispersionDMX
+
+def test_startMJD_greaterThan_endMJD():
+    """ Check for error when start MJD is larger than end MJD. """
+    with pytest.raises(ValueError):
+        dm_mod=DispersionDMX()
+        dm_mod.add_DMX_range(58100,58000,3,1,frozen=False)
+
+def test_only_one_MJD():
+    """ Check for error when one of the MJDs is None and the other isn't. """ 
+    with pytest.raises(ValueError):
+        dm_mod=DispersionDMX()
+        dm_mod.add_DMX_range(None,58000,3,1,frozen=False)
+    with pytest.raises(ValueError):
+        dm_mod=DispersionDMX()
+        dm_mod.add_DMX_range(58000,None,3,1,frozen=False)
+
+def test_duplicate_index()
+    """ Check for error when a duplicate DMX index is used. """ 
+    with pytest.raises(ValueError):
+        dm_mod=DispersionDMX()
+        dm_mod.add_DMX_range(58000,58100,3,1,frozen=False)
+        dm_mod.add_DMX_range(58200,58300,3,1,frozen=False)
+
+def test_remove_nonexistent_index()
+    """ Check for error when a unused DMX index is removed. """
+    with pytest.raises(ValueError):
+        dm_mod=DispersionDMX()
+        dm_mod.remove_DMX_range(58000,58100,3,1,frozen=False)
+
+def test_add_DMX():
+    """ Check that the DMX model contains the DMX events after they are added. """
+    dm_mod=DispersionDMX()
+    dm_mod.add_DMX_range(58000,58100,3,1,frozen=False)
+    assert len(dm_mod.params) == 7
+
+def test_remove_DMX():
+    """Check the different ways of grouping components."""
+    dm_mod=DispersionDMX()
+    dm_mod.add_DMX_range(58000,58100,3,1,frozen=False)
+    dm_mod.remove_DMX_range(3)
+    assert len(dm_mod.params) == 4
+
```

Comment:
What happens if someone removes the spurious DMX range that was added when the object was created? Can you still add DMX ranges? Can it still compute delays?

Reply:
yes, you can. there will still be an element in the component's params called 'DMX', but the 'DMX_0001' component can be removed with no problem - and more can be added afterwards.